.github/workflows/branch_to_staging.yml

```diff
@@ -0,0 +1,21 @@
+name: Build/Deploy Branch to Staging
+on:
+  workflow_dispatch:
+jobs:
+  test:
+    uses: ./.github/workflows/ci.yml
+    secrets: inherit
+  call_build_and_push:
+    needs: test
+    uses: ./.github/workflows/build.yml
+    with:
+      image_name: ${{ github.repository }}
+      image_tag: main
```

Comment:
Seems like the tag needs to be based on the branch, not main

Reply:
Great catch.  Done.